<a href="https://colab.research.google.com/github/NakulDeotale/FlyrankML/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [1]:
print("I choose freestyle: Growth/Recovery Prediction. I want to predict whether a page will decline or recover in the next 30 days, using features from the prior 90 days. Since editors can't review every page. A prediction gives them a prioritized list of which pages to check first, saving time and catching problems earlier.")

I choose freestyle: Growth/Recovery Prediction. I want to predict whether a page will decline or recover in the next 30 days, using features from the prior 90 days. Since editors can't review every page. A prediction gives them a prioritized list of which pages to check first, saving time and catching problems earlier.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [2]:
print('1. It saves time and promotes efficiency by first reviewing the pages that are likely to decline and need careful attention to recover, to prevent them from declining.')
print('2. The page editors will act on it.')
print('3. A wrong recommendation may lead to the waste of time in reviewing healthy page and a declining page being left out.')

1. It saves time and promotes efficiency by first reviewing the pages that are likely to decline and need careful attention to recover, to prevent them from declining.
2. The page editors will act on it.
3. A wrong recommendation may lead to the waste of time in reviewing healthy page and a declining page being left out.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
import pandas as pd

df = pd.read_csv('/content/content_refresh_anonymized.csv')

# What % of pages are declining?
df['trend_direction'].value_counts(normalize=True)*100

,proportion
trend_direction,
down,54.206667
stable,19.873333
up,14.626667
new,7.453333
flat,3.840000


######54.2% of pages are declining

In [4]:
# How many pages have enough traffic to matter?
(df['impressions_90d'] >= 100).mean()*100

np.float64(73.35333333333334)

######On an average 73.35% of pages have enough traffic, which tells us that the problem is worth solving.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [5]:
print('I can claim (observed, directional, decision-support):')
print("Pages our model flagged as high-risk were more likely to decline in the next 30 days than lower-ranked pages - if the validation is clean and windows don't overlap.")
print("Our ranked queue would surface different candidates than a plain stale-age rule - by comparing overlap.")
print("Of the top 50 pages flagged, we estimate X% showed a decline in the following month - with a pre-registered metric and honest threshold.")
print("")
print('I cannot claim:')
print("Editing these pages caused them to recover — correlation is not causation; no experiment, no causal claim.")
print("This predicts Google algorithm updates — the data measures outcomes, not Google's ranking reasons.")
print("Results generalize to all clients or time periods — findings depend on the client mix, time windows, and thresholds chosen.")

I can claim (observed, directional, decision-support):
Pages our model flagged as high-risk were more likely to decline in the next 30 days than lower-ranked pages - if the validation is clean and windows don't overlap.
Our ranked queue would surface different candidates than a plain stale-age rule - by comparing overlap.
Of the top 50 pages flagged, we estimate X% showed a decline in the following month - with a pre-registered metric and honest threshold.

I cannot claim:
Editing these pages caused them to recover — correlation is not causation; no experiment, no causal claim.
This predicts Google algorithm updates — the data measures outcomes, not Google's ranking reasons.
Results generalize to all clients or time periods — findings depend on the client mix, time windows, and thresholds chosen.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.